include/zephyr/sys/crc.h

```diff
@@ -263,6 +265,32 @@ uint32_t crc32_ieee_update(uint32_t crc, const uint8_t *data, size_t len);
 uint32_t crc32_c(uint32_t crc, const uint8_t *data,
 		 size_t len, bool first_pkt, bool last_pkt);
 
+/**
+ * @brief Update a CRC-32K/4.2 (*op) (Koopman) checksum. This is a good HD=4
+ * checksum up to 2,147,483,615 bits and HD=5/6 up to 6,167 bits.
+ *
+ * Hamming Distance and properties:
+ *
+ * - Polynomial: 0x93a409eb
+ * - reflect-in: false
+ * - initial value (xor-in): provided by caller as crc argument (0xFFFFFFFF is OK)
+ * - reflect-out: false
+ * - xor-out: 0
+ * - HD=4 @ 2,147,483,615 bits
+ * - HD=5 @ 6,167 bits
+ * - HD=6 @ 6,167 bits
+ * - HD=7 @ 148 bits
+ *
+ * Reference: https://users.ece.cmu.edu/~koopman/crc/crc32.html
+ *
+ * @param crc       CRC32 checksum that needs to be updated.
+ * @param data      Pointer to data on which the CRC should be calculated.
+ * @param len       Data length.
+ *
+ * @return CRC32 value.
+ */
+uint32_t crc32_k_4_2_update(uint32_t crc, uint8_t const *data, size_t len);
```

Comment:
Put `const` modifiers before the type, to keep it consistent. (Here and other new code)